x-pack/plugins/infra/server/lib/alerting/inventory_metric_threshold/lib/create_condition_script.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { Comparator } from '../../../../../common/alerting/metrics';
+import { SnapshotMetricType } from '../../../../../common/inventory_models/types';
+import { convertMetricValue } from './convert_metric_value';
+
+export const createConditionScript = (
```

Comment:
Probably premature optimization, but this got me thinking if we should maybe pre-register this as a stored script instead (https://www.elastic.co/guide/en/elasticsearch/reference/current/create-stored-script-api.html)

We'd avoid complication cost on every query and it might be useful to have "the way" to register utility scripts as part of kibana startup.

Reply:
I doubt the compilation cost is worth optimizing for since it would add addition complexity of having to deal the stored scripts.